zkevm-circuits/src/bytecode_circuit/bytecode_unroller.rs

```diff
@@ -212,17 +202,20 @@ impl<F: Field> Config<F> {
                 ),
             );
             cb.require_equal(
-                "if row.tag == tag.Length: value == row.hash_length",
+                "if row.tag == tag.Length: value == row.code_length",
                 meta.query_advice(value, Rotation::cur()),
-                meta.query_advice(hash_length, Rotation::cur()),
+                meta.query_advice(code_length, Rotation::cur()),
             );
-            cb.condition(length_is_zero.clone().is_zero_expression, |cb| {
-                cb.require_equal(
-                    "if length == 0: hash == RLC(EMPTY_HASH, randomness)",
-                    meta.query_advice(hash, Rotation::cur()),
-                    Expression::Constant(keccak(&[], r)),
-                );
-            });
+            // FIXME: Since randomness is only known at synthesis time, the RLC of empty
+            // code_hash is not constant.  Consider doing a lookup to the empty code_hash
+            // value? cb.condition(length_is_zero.clone().is_zero_expression,
+            // |cb| {     cb.require_equal(
+            //         "if length == 0: code_hash == RLC(EMPTY_HASH, randomness)",
+            //         meta.query_advice(bytecode_table.code_hash, Rotation::cur()),
+            //         Expression::Constant(keccak(&[], randomness)),
+            //     );
+            // });
+
```

Comment:
Ah, but it's not that easy!  This gate needs to be redesigned.  Currently this gate requires the randomness to be constant, as it expects the `RLC(keccak([]))` to be a constant expression, but this won't be the case, because the randomness is different for each proof.

So instead we need to hardcode the bytes of `keccak([])` and then build the RLC expression.  Actually, after explaining the solution it sounds easy to do.  I think I can update the PR with this.

Reply:
Ah right, it's not that simple to just uncomment, the approach you describe sounds good to me, we can also leave it until challenge API is ready.